File: src/main/java/rs/ac/uns/ftn/svt/model/Facility.java
package rs.ac.uns.ftn.svt.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@JsonIgnoreProperties({"reviews", "exercises", "manages", "images", "disciplines", "workDays"})
public class Facility {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    private LocalDate createdAt;

    private String address;

    private String city;

    private Double totalRating;

    private boolean active;

    @OneToMany(mappedBy = "facility")
    private List<Review> reviews;

    @OneToMany(mappedBy = "facility")
    private List<Exercise> exercises;

    @OneToMany(mappedBy = "facility")
    private List<Manages> manages;

    @OneToMany(mappedBy = "facility")

    private List<Image> images;

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Discipline> disciplines = new ArrayList<>();

    @OneToMany(mappedBy = "facility", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<WorkDay> workDays = new ArrayList<>();

}

File: kvt-projekat2024/src/app/models/Facility.ts
import { Review } from "./Review";
import { Exercise } from "./Exercise";
import { Manages } from "./Manages";
import { Image } from "./Image";
import { Discipline } from "./Discipline";
import { WorkDay } from "./WorkDay";

export class Facility {
  id: number;
  name: string;
  description: string;
  createdAt: Date;
  address: string;
  city: string;
  totalRating: number;
  active: boolean;
  reviews: Review[];
  exercises: Exercise[];
  manages: Manages[];
  images: Image[];
  disciplines: Discipline[];
  workDays: WorkDay[];

  constructor(
    id: number,
    name: string,
    description: string,
    createdAt: Date,
    address: string,
    city: string,
    totalRating: number,
    active: boolean,
    reviews: Review[],
    exercises: Exercise[],
    manages: Manages[],
    images: Image[],
    disciplines: Discipline[],
    workDays: WorkDay[]
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.createdAt = createdAt;
    this.address = address;
    this.city = city;
    this.totalRating = totalRating;
    this.active = active;
    this.reviews = reviews;
    this.exercises = exercises;
    this.manages = manages;
    this.images = images;
    this.disciplines = disciplines;
    this.workDays = workDays;
  }
}
